app/src/main/java/ch/epfl/sweng/favors/database/User.java

```diff
@@ -4,27 +4,46 @@
 
 public class User extends DatabaseHandler {
     private static final String COLLECTION = "users";
+    private FirebaseAuth instance;
 
     private static User user = new User();
     public static User getMain(){
         return user;
     }
 
+    public FirebaseAuth getInstance() {
+        return instance;
+    }
+
     public static void setMain(String id){
         user = new User(id);
     }
+    public static void setMain(User u) {user = u; }
 
     public enum StringFields implements DatabaseStringField{firstName, lastName, email, sex, basedLocation}
 
     public User(){
         super(StringFields.values(), COLLECTION,FirebaseAuth.getInstance().getUid());
-        if(FirebaseAuth.getInstance().getUid() != null)
+        instance = FirebaseAuth.getInstance();
+        if(instance.getUid() != null){
             updateFromDb();
+        }
+
     }
 
     public User(String id){
         super(StringFields.values(), COLLECTION,id);
-        if(FirebaseAuth.getInstance().getUid() != null)
+        instance = FirebaseAuth.getInstance();
+        if(instance.getUid() != null) {
+            updateFromDb();
+        }
+    }
+
+    public User(FirebaseAuth instance){
+        super(StringFields.values(), COLLECTION,instance.getUid());
+        this.instance = instance;
+        if(instance.getUid() != null) {
             updateFromDb();
+        }
```

Comment:
Did you modify the User class? Or is this a because you pulled another branch into yours?

Reply:
Yes we slightly modified it : 
Added a constructor which takes a FirebaseAuth in argument
Added the setMain(User u) method

Reply:
Ok. You can tell me tomorrow why you did that.